docs: Add USBGuard installation and configuration guide (#85)
ni/nilrt-docs: docs/source/usbguard/usbguard.rst

such as ID, serial number, and more.

The primary purpose of USBGuard is to defend against threats like BadUSB attacks, where compromised
USB devices can be used to exploit or take control of a computer. By enforcing strict authorization
policies, USBGuard helps prevent rogue devices from gaining access, reducing the risk of data theft,
malware infection, or system compromise.

You might want to use USBGuard if you need to:
- Secure sensitive systems against unauthorized USB access
- Prevent data exfiltration via removable media
- Limit the risk of malware introduced through USB devices
- Maintain compliance with security policies that restrict external device usage

USBGuard is especially valuable in environments where physical access to computers is possible, such as
offices, labs, or public spaces. Its flexible rule language and integration with Linux kernel features make
it a powerful tool for enhancing endpoint security.

Installation
============

To install USBGuard on an NI Linux Real-Time system, simply run the following command
from your target:

.. code:: bash
    
    opkg install usbguard

This will install, but not enable, USBGuard on your system.

> The default rule file is empty, and all USB devices will be blocked until you create rules to allow

Reviewer: I'm talking about the paragraph about the default rule file, in case that's not clear.

> The default rule file is empty, and all USB devices will be blocked until you create rules to allow

Author: I put it here to make clear that if you start it without configuring it, you may lock yourself out of your system until you reboot. I didn't want to move the section that expands on it up any higher in the document. I think it works either place so moved.

This will create a rules.conf file that whitelists all devices currently connected to your system. You
can then edit this file to fine-tune your device policies.

Once you are satisfied with your rules, you can enable USBGuard to run at startup by running:

.. code:: bash

    # Force remove any existing symlinks
    update-rc.d -f usbguard remove
    # Add USBGuard to startup, starting if applicable for current runlevel, using
    # defaults from the init script
    update-rc.d -s usbguard defaults

Service Management
==================

With sysvinit, you can manually control the USBGuard service using the following commands:

.. code:: bash

    service usbguard start      # Start the USBGuard service
    service usbguard stop       # Stop the USBGuard service
    service usbguard restart    # Restart the USBGuard service
    service usbguard status     # Check the status of the USBGuard service

These commands require root privileges.

Security Warning
================

Be aware that enabling USBGuard without a proper ruleset may block all USB devices, including
critical input devices such as your keyboard and mouse. Always ensure your rules.conf file

> authorizes the devices you need before starting or enabling the service.

Author: Done, and updated screenshot in PR description

malicious USB devices. It provides a robust mechanism for controlling which USB devices are allowed to
connect to your system by implementing whitelisting and blacklisting policies based on device attributes
such as ID, serial number, and more.

The primary purpose of USBGuard is to defend against threats like BadUSB attacks, where compromised
USB devices can be used to exploit or take control of a computer. By enforcing strict authorization
policies, USBGuard helps prevent rogue devices from gaining access, reducing the risk of data theft,
malware infection, or system compromise.

You might want to use USBGuard if you need to:
- Secure sensitive systems against unauthorized USB access
- Prevent data exfiltration via removable media
- Limit the risk of malware introduced through USB devices
- Maintain compliance with security policies that restrict external device usage

USBGuard is especially valuable in environments where physical access to computers is possible, such as
offices, labs, or public spaces. Its flexible rule language and integration with Linux kernel features make
it a powerful tool for enhancing endpoint security.

Installation
============

> Support for USBGuard is available on the NI Linux RT System
> Image starting with version 11.3 (build 25.8.0).

Reviewer: ```suggestion
USBGuard requires the NI Linux RT System
Image version 11.3 (build 25.8.0) or newer.
```

"available on" makes it sound like usbguard is included in the BSI

Author: Changed

To install USBGuard on an NI Linux Real-Time system, simply run the following command
from your target:

.. code:: bash
    
    opkg install usbguard

This will install, but not enable, USBGuard on your system.

You can check the status of the USBGuard service with:

.. code:: bash

    service usbguard status

Configuration
=============

USBGuard uses configuration files to control its behavior and define which USB devices are allowed or
blocked. The two main files are:

> The default rule file is empty, and all USB devices will be blocked until you create rules to allow

Reviewer: You inserted this paragraph into the middle of a sentence.
